Mathlib/Algebra/Module/Submodule/IterateMapComap.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2024 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.Algebra.Module.Submodule.Ker
+
+/-!
+
+# Iterate maps and comaps of submodules
+
+Some preliminary work for establishing the strong rank condition for noetherian rings.
+
+Given two linear maps `f i : M →ₗ[R] M₂` and a submodule `K : Submodule R M`, we can define
```

Comment:
Is there a particular reason that you call the modules `M` and `M₂`? I would prefer a bit more symmetry. Either `M` and `N`, or give both a subscript.